packages/sidecar/src/main.ts

```diff
@@ -209,6 +210,34 @@ function handleClearRequest(req: IncomingMessage, res: ServerResponse): void {
   }
 }
 
+function openRequestHandler(req: IncomingMessage, res: ServerResponse) {
+  // We're only interested in handling a POST request
+  if (req.method !== 'POST') {
+    res.writeHead(405);
+    res.end();
+    return;
+  }
+
+  let requestBody = '';
+  req.on('data', chunk => {
+    requestBody += chunk;
+  });
+
+  req.on('end', () => {
+    launchEditor(
+      // filename:line:column
+      // both line and column are optional
+      requestBody,
```

Comment:
I think this is not working.

As in stacktrace we didn't get the absolute path of the file.
For example: 
In my usecase i am getting filename from sentry sdk as "components/Navbar.tsx"

And we are sending the same filename with line and colno to launchEditor and it is not opening the code editor.

Ideally launchEditor needs absolute path - which is my case is "/home/newuser/Documents/shubhdeepchhabra.in/components/Navbar.tsx"

let me know if am wrong somewhere.

Reply:
So I was testing this with the built-in Astro project with `pnpm dev:playground`. There, the stacktrace filename gets converted into an absolute path through the `contextlines` endpoint and that's why this worked for me locally.

I now added logic to `resolve` relative paths based on a provided `basePath` like the static file serve handler or relative to where the process was started which should be the project base path ideally.

If you can provide me a repro case, I'd love to make this more robust.

Reply:
Yeah, we can do this. 

- But basePath is optional and undefined by default unless user pass it explicitly.
  - there are 3-4 ways to run sidecar server, we need to handle all of them.
- And, i think this will also cause issue -> rn sidecar only run once on a machine and handle multiple instances. For example -  i can run multiple spotlight overlay instances with single sidecar instance. how we would get diff basepath.

Reply:
>But basePath is optional and undefined by default unless user pass it explicitly. there are 3-4 ways to run sidecar server, we need to handle all of them.

I think the default of `process.cwd()` covers most cases at the moment.

>And, i think this will also cause issue -> rn sidecar only run once on a machine and handle multiple instances. For example - i can run multiple spotlight overlay instances with single sidecar instance. how we would get diff basepath.

Agreed. This should be handled from the front-end/overlay side tho, ensuring absolute URLs if/when possible. This may not always be available to us but I'd rather deal with this as it comes through user-feedback.

Reply:
Yes agreed!
we can go ahead now with default value